src/main/java/org/snomed/snowstorm/core/data/services/servicehook/CommitInformation.java

```diff
@@ -7,21 +7,33 @@
 
 public class CommitInformation {
 
-	private final String path;
+	private final String sourceBranchPath;
+	private final String targetBranchPath;
 	private final Commit.CommitType commitType;
 	private final long headTime;
 	private final Map<String, Object> metadata;
 
 	public CommitInformation(Commit commit) {
 		final Branch branch = commit.getBranch();
-		path = branch.getPath();
 		commitType = commit.getCommitType();
 		headTime = commit.getBranch().getHeadTimestamp();
 		metadata = branch.getMetadata().getAsMap();
+
+		if (Commit.CommitType.CONTENT == commitType) {
+			this.sourceBranchPath = branch.getPath();
+			this.targetBranchPath = null;
+		} else {
+			this.sourceBranchPath = commit.getSourceBranchPath();
+			this.targetBranchPath = branch.getPath();
+		}
+	}
+
+	public String getSourceBranchPath() {
+		return sourceBranchPath;
```

Comment:
The `sourceBranchPath` and `targetBranchPath` come from the same location for both `Rebase` and `Promotion`, however, are different when `Content` is the `CommitType`.